File: searchInsertPosition/schinpos.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        int pos=0;
        for(int n:nums){
            if(n<target){
                pos++;
            }else{
                break;
            }
        }
        return pos;
    }
}

class Solution {
    public int searchInsert(int[] nums, int target) {
        int begin = 0;
        int end = nums.length-1;
        while(begin<=end){
            int mid = (begin+end)/2;
            if(nums[mid]>target){
                end = mid-1;
            }else if(nums[mid]<target){
                begin=mid+1;
            }
            else{
                return mid;
            }
        }
        return begin;
    }
}

File: searchInsertPosition/schinpos.py
class Solution:
    def searchInsert(self, nums: 'List[int]', target: 'int') -> 'int':
        pos=0
        for n in nums:
            if(n<target):
                pos+=1
        return pos

class Solution:
    def searchInsert(self, nums: 'List[int]', target: 'int') -> 'int':
        begin,end=0,len(nums)-1
        while(begin<=end):
            mid = int((begin+end)/2)
            if nums[mid]==target:
                return mid
            elif nums[mid]>target:
                end = mid-1
            else:
                begin = mid+1
        return begin